ROX-23786: cosign certificate verification

dhaus67 opened the pull request:
Version(s):
Latest ACS version (4.5)

Issue:
https://issues.redhat.com/browse/ROX-23786

Link to docs preview:
https://78177--ocpdocs-pr.netlify.app/openshift-acs/latest/operating/verify-image-signatures.html

Additional information:
Docs for the newly introduced verification method using cosign certificates. This is essentially only an extension within the Signature integration.

dhaus67 commented:
/label peer-review-needed

kcarmichael08 commented on modules/configure-signature-integration.adoc:
```diff
@@ -5,7 +5,13 @@
 = Configuring signature integration
 
-Before performing image signature verification, you must first add your Cosign public keys in {product-title-short}.
+Before performing image signature verification, you must first create a Signature integration in {product-title-short}.
```
```suggestion
Before performing image signature verification, you must first create a signature integration in {product-title-short}.
```

kcarmichael08 commented on operating/verify-image-signatures.adoc:
```diff
@@ -13,11 +13,12 @@ You can also enforce the policy by using the {product-title-short} admission con
 ====
-* {product-title-short} 3.70 only supports Cosign signatures and Cosign public key signature verification. For more information about Cosign, see link:https://docs.sigstore.dev/cosign/overview[Cosign overview].
-* You must configure signature integration with at least 1 Cosign public key for signature verification.
+* {product-title-short} only supports Cosign signatures and Cosign Public Keys / Certificates verification. For more information about Cosign, see link:https://docs.sigstore.dev/cosign/overview[Cosign overview].
```
```suggestion
* {product-title-short} only supports Cosign signatures and Cosign Public Keys/Certificates verification. For more information about Cosign, see link:https://docs.sigstore.dev/cosign/overview[Cosign overview].
```

kcarmichael08 commented on operating/verify-image-signatures.adoc:
```diff
@@ -13,11 +13,12 @@ You can also enforce the policy by using the {product-title-short} admission con
-* {product-title-short} 3.70 only supports Cosign signatures and Cosign public key signature verification. For more information about Cosign, see link:https://docs.sigstore.dev/cosign/overview[Cosign overview].
-* You must configure signature integration with at least 1 Cosign public key for signature verification.
+* {product-title-short} only supports Cosign signatures and Cosign Public Keys / Certificates verification. For more information about Cosign, see link:https://docs.sigstore.dev/cosign/overview[Cosign overview].
+* For Cosign signature verification, {product-title-short} does not reach out to the transparency log link:https://docs.sigstore.dev/logging/overview/[Rekor].
```
I would change "reach out to" to "communicate with" or "contact" or "get data from" or something else.

kcarmichael08 commented on modules/configure-signature-integration.adoc:
```diff
@@ -17,8 +23,28 @@ Before performing image signature verification, you must first add your Cosign p
+. Click *Cosign certificates* -> *Add a new certificate verification*.
+. Enter the *Certificate OIDC Issuer*. You can optionally use regular expressions in link:https://github.com/google/re2/wiki/Syntax[RE2 Syntax].
+. Enter the *Certificate identity*. You can optionally use regular expressions in link:https://github.com/google/re2/wiki/Syntax[RE2 Syntax].
+. (Optional) You can add the PEM-encoded certificate chain to verify certificates. If no chain is provided, certificates will be verified against the link:https://docs.sigstore.dev/certificate_authority/overview/[Fulcio] root.
```
```suggestion
. (Optional) You can add the PEM-encoded certificate chain to verify certificates. If no chain is provided, certificates are verified against the link:https://docs.sigstore.dev/certificate_authority/overview/[Fulcio] root.
```

kcarmichael08 commented on modules/configure-signature-integration.adoc:
```diff
@@ -5,7 +5,13 @@
+Before performing image signature verification, you must first create a Signature integration in {product-title-short}.
+
+A signature integration can be configured with multiple verification methods. The following verification methods are supported:
+* Cosign public keys.
```
```suggestion
* Cosign public keys
```

kcarmichael08 commented on modules/configure-signature-integration.adoc:
```diff
@@ -5,7 +5,13 @@
+
+A signature integration can be configured with multiple verification methods. The following verification methods are supported:
+* Cosign public keys.
+* Cosign certificates.
```
```suggestion
* Cosign certificates
```

kcarmichael08 commented on modules/configure-signature-integration.adoc:
```diff
@@ -17,8 +23,28 @@ Before performing image signature verification, you must first add your Cosign p
+
+.Procedure
+
+. In the {product-title-short} portal, select *Platform Configuration* → *Integrations*.
```
```suggestion
. In the {product-title-short} portal, select *Platform Configuration* -> *Integrations*.
```
We need to use `->` so that it transforms properly in our doc publishing.

kcarmichael08 commented on modules/configure-signature-integration.adoc:
```diff
@@ -17,8 +23,28 @@ Before performing image signature verification, you must first add your Cosign p
+.Procedure
+
+. In the {product-title-short} portal, select *Platform Configuration* → *Integrations*.
+. Scroll down to the *Signature Integrations* section and click *Signature*.
```
```suggestion
. Scroll to *Signature Integrations* and click *Signature*.
```

kcarmichael08 commented on modules/configure-signature-integration.adoc:
```diff
@@ -17,8 +23,28 @@ Before performing image signature verification, you must first add your Cosign p
 . (Optional) You can add more than one key by clicking *Add a new public key* and entering the details.
 . Click *Save*.
+
+== Configuring Cosing Certificates
```
```suggestion
== Configuring Cosign Certificates
```

kcarmichael08 commented on modules/configure-signature-integration.adoc:
```diff
@@ -17,8 +23,28 @@ Before performing image signature verification, you must first add your Cosign p
+
+.Prerequisites
+
+* You must already have the certificate identity, and issuer. Optionally, you also need a PEM-encoded certificate and chain. For more information about Cosign Certificates, see link:https://docs.sigstore.dev/verifying/verify/#verify-image-with-user-provided-trusted-chain[Cosign Certificates]
```
```suggestion
* You must already have the certificate identity and issuer. Optionally, you also need a PEM-encoded certificate and chain. For more information about Cosign Certificates, see link:https://docs.sigstore.dev/verifying/verify/#verify-image-with-user-provided-trusted-chain[Cosign Certificates]
```

kcarmichael08 reviewed (commented):
I marked a few style issues and a typo but it looks really good. Thank you!!!

dhaus67 commented:
@kcarmichael08 addressed the feedback, I've also taken the liberty to:
- adjust some style issues in the previous, already existing section.
- lower-cased usages of `public keys` + `certificates`.
- use the field names within the `Optional` sections of adding the certificate / certificate chain for `Cosign Certificates`.

Hope that makes sense.

kcarmichael08 commented on modules/configure-signature-integration.adoc:
```diff
@@ -5,20 +5,46 @@
+Before performing image signature verification, you must first create a signature integration in {product-title-short}.
+
+A signature integration can be configured with multiple verification methods. The following verification methods are supported:
+* Cosign public keys
```
I think you need a blank line here so the bullets look OK:
![Screenshot 2024-06-27 at 1 48 17 PM](https://github.com/openshift/openshift-docs/assets/96192119/a2356549-4789-4110-a6e1-736256a72a91)

kcarmichael08 reviewed (commented):
Thanks for going back and making the other changes! Looks good with one small issue.

kcarmichael08 reviewed (commented):
I'm sorry, I realized that the headings needed the Anchor IDs above them. Then I think it should be OK to merge.

https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#anchoring-in-module-files

kcarmichael08 commented:
/cherrypick rhacs-docs-4.5
